### PR TITLE
Rename WordPressEManga to WordPressMangastream

### LINKED ISSUE #2302: [Asura Scans] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

When Refresh Manga List Popup Message
Failed to receive content from "https://asurascans.com/wp-admin/admin-ajax.php" (status: 400) - 

1. Asura Scans
4. Error: Failed to receive content from "https://asurascans.com/wp-admin/admin-ajax.php" (status: 400) - 
    at Connector.mjs:453
    at async AsuraScans._getMangasFromPage (WordPressMadara.mjs:46)
    at async AsuraScans._getMangas (WordPressMadara.mjs:38)
    at async AsuraScans._getMangaList (Connector.mjs:557) AsuraScans {id: "asurascans", label: "Asura Scans", tags: Array(2), isLocked: false, initialized: true, …}

Screenshot (3)

 - Version 6.17

### PR DESCRIPTION
**Pull request by XenonR**
Add new WP template for "Mangastream"
Also fixes: #2302

### CODE REVIEW

**Review by 27136773 — COMMENTED** [review]
The template is already present, its available under its old alias `WordPressEManga`
I recommend to use the existing template instead of adding a duplicate.

Template:
https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/templates/WordPressEManga.mjs
Sample:
https://bacamanga.co/manga/

**Comment by 27136773** [comment]
May also extend the existing template with `getMangaFromURI` method

**Comment by XenonR** [comment]
Looks a bit messy now, but i just renamed `WordPressEmanga` to `WordpressMangastream` and added the method `_getMangaFromURI`. Also modified `this.querMangaTitleFromURI` slightly to be more 'broad'. Did full testing with AsuraScans and BacaKomik.

File: src/web/mjs/connectors/DoujinDesu.mjs [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import WordPressEManga from './templates/WordPressEManga.mjs';
+import WordPressMangastream from './templates/WordWordPressMangastreamPressEManga.mjs';
```
Comment (27136773) [inline src/web/mjs/connectors/DoujinDesu.mjs]: Invalid template file name

File: src/web/mjs/connectors/KomikIndoWeb.mjs [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import WordPressEManga from './templates/WordPressEManga.mjs';
+import WordPressMangastream from './templates/WordPressWordPressMangastreamEManga.mjs';
```
Comment (27136773) [inline src/web/mjs/connectors/KomikIndoWeb.mjs]: Invalid template file

File: src/web/mjs/connectors/templates/WordPressMangastream.mjs [hunk 3]
```diff
@@ -22,7 +24,7 @@ export default class WordPressEManga extends Connector {
         return data.map(element => {
             return {
                 id: this.getRootRelativeOrAbsoluteLink(element, request.url),
-                title: element.text.trim()
+                title: element.title.trim()
```
Comment (27136773) [inline src/web/mjs/connectors/templates/WordPressMangastream.mjs]: Is it safe to assume all websites based on this template have a `title` attribute set on the manga link?

**Review by 27136773 — CHANGES_REQUESTED** [review]
Some imports needs to be fixed.

Check if any other Connector failed to load during HakuNeko startup by going carefully through the warnings and errors in developer console.

File: src/web/mjs/connectors/templates/WordPressMangastream.mjs (hunk 3 above)
Comment (XenonR) [reply src/web/mjs/connectors/templates/WordPressMangastream.mjs]: No if i assume it worked OK before. I changed it to `title: element.title ? element.title.trim() : element.text.trim()` to be sure.

File: src/web/mjs/connectors/templates/WordPressMangastream.mjs [hunk 4]
```diff
@@ -22,7 +24,7 @@ export default class WordPressEManga extends Connector {
         return data.map(element => {
             return {
                 id: this.getRootRelativeOrAbsoluteLink(element, request.url),
-                title: element.text.trim()
+                title: element.title ? element.title.trim() : element.text.trim()
```
Comment (27136773) [inline src/web/mjs/connectors/templates/WordPressMangastream.mjs:27]: Clever solution, well done

**Review by 27136773 — APPROVED** [review]
Looks good now
